internal/server/config.go

```diff
@@ -678,23 +548,34 @@ func loadConfig(db *gorm.DB, config Config) error {
 }
 
 func loadProviders(db *gorm.DB, providers []Provider) error {
-	toKeep := make([]uid.ID, 0)
+	keep := make([]uid.ID, 0, len(providers)+1)
```

Comment:
We should make this `+1` a named constant rather than a magic number

Reply:
Wasn't a fan of this either. Instead, internal configurations will be injected into the config at an earlier stage so this won't be necessary anymore